src/pybind/mgr/dashboard_v2/controllers/dashboard.py

```diff
@@ -0,0 +1,128 @@
+        }
+
+    @cherrypy.expose
+    @cherrypy.tools.json_out()
+    def health(self):
+        # Fuse osdmap with pg_summary to get description of pools
+        # including their PG states
+
+        osd_map = self.osd_map()
+
+        pg_summary = self.mgr.get("pg_summary")
+
+        pools = []
+
+        pool_stats = defaultdict(lambda: defaultdict(
+            lambda: collections.deque(maxlen=10)))
+
+        df = self.mgr.get("df")
+        pool_stats_dict = dict([(p['id'], p['stats']) for p in df['pools']])
+        now = time.time()
+        for pool_id, stats in pool_stats_dict.items():
+            for stat_name, stat_val in stats.items():
+                pool_stats[pool_id][stat_name].appendleft((now, stat_val))
+
+        for pool in osd_map['pools']:
+            pool['pg_status'] = pg_summary['by_pool'][pool['pool'].__str__()]
+            stats = pool_stats[pool['pool']]
+            s = {}
+
+            def get_rate(series):
+                if len(series) >= 2:
+                    return (float(series[0][1]) - float(series[1][1])) / (float(series[0][0]) - float(series[1][0]))
+                else:
+                    return 0
+
+            for stat_name, stat_series in stats.items():
+                s[stat_name] = {
+                    'latest': stat_series[0][1],
+                    'rate': get_rate(stat_series),
+                    'series': [i for i in stat_series]
+                }
+            pool['stats'] = s
+            pools.append(pool)
+
+        # Not needed, skip the effort of transmitting this
+        # to UI
+        del osd_map['pg_temp']
+
+        df['stats']['total_objects'] = sum(
+            [p['stats']['objects'] for p in df['pools']])
+
+        return {
+            "health": self.health_data(),
+            "mon_status": self.mon_status(),
+            "fs_map": self.mgr.get('fs_map'),
+            "osd_map": osd_map,
+            # "clog": list(self.mgr.log_buffer),
+            # "audit_log": list(self.mgr.audit_buffer),
+            "pools": pools,
+            "mgr_map": self.mgr.get("mgr_map"),
+            "df": df
+        }
+
+    def mon_status(self):
+        mon_status_data = self.mgr.get("mon_status")
+        return json.loads(mon_status_data['json'])
+
+    def osd_map(self):
+        osd_map = self.mgr.get("osd_map")
+
+        assert osd_map is not None
+
+        osd_map['tree'] = self.mgr.get("osd_map_tree")
+        osd_map['crush'] = self.mgr.get("osd_map_crush")
+        osd_map['crush_map_text'] = self.mgr.get("osd_map_crush_map_text")
+        osd_map['osd_metadata'] = self.mgr.get("osd_metadata")
+
+        return osd_map
+
+    def health_data(self):
+        health_data = self.mgr.get("health")
+        health =  json.loads(health_data['json'])
+
+        # Transform the `checks` dict into a list for the convenience
+        # of rendering from javascript.
+        checks = []
+        for k, v in health['checks'].iteritems():
```

Comment:
Please change to: `for k, v in health['checks'].items():` (Python 3 compatibility)

Reply:
done